62.CAD/common.hlsl

```diff
@@ -148,9 +131,65 @@ struct PSInput
     void setBezierP0(float2 p0) { data2.xy = p0; }
     void setBezierP1(float2 p1) { data2.zw = p1; }
     
-    // data3 (zw reserved for later)
+    // data3 xy
     float2 getBezierP2() { return data3.xy; }
     void setBezierP2(float2 p2) { data3.xy = p2; }
+
+    // Curves are split in the vertex shader based on their tmin and tmax
+    // Min curve is smaller in the minor coordinate (e.g. in the default of y top to bottom sweep,
+    // curveMin = smaller x / left, curveMax = bigger x / right)
+    // TODO: possible optimization: passing precomputed values for solving the quadratic equation instead
+
+    // data2, data3, data4
+    float2 getCurveMinA() { return data2.xy; }
+    float2 getCurveMinB() { return data2.zw; }
+    float2 getCurveMinC() { return data3.xy; }
+    float2 getCurveMaxA() { return data3.zw; }
+    float2 getCurveMaxB() { return data4.xy; }
+    float2 getCurveMaxC() { return data4.zw; }
+    
+    void setCurveMinA(float2 p) { data2.xy = p; }
+    void setCurveMinB(float2 p) { data2.zw = p; }
+    void setCurveMinC(float2 p) { data3.xy = p; }
+    void setCurveMaxA(float2 p) { data3.zw = p; }
+    void setCurveMaxB(float2 p) { data4.xy = p; }
+    void setCurveMaxC(float2 p) { data4.zw = p; }
```

Comment:
These should be floats instead of float2s, you only need to evaluate at minor axis.
rename to `set(Max/Min)CurveMinorAxisQuadratic(nbl::hlsl::equations::Quadratic)`
and make adjustments on which internal data to use.